xrpl_binary_codec/src/parser.rs

```diff
@@ -0,0 +1,616 @@
+use xrpl_types::{Amount, Memo, Transaction, TransactionType};
+
+use crate::{
+    error::BinaryCodecError,
+    serializer::{HASH_PREFIX_TRANSACTION, HASH_PREFIX_UNSIGNED_TRANSACTION_SINGLE},
+};
+
+#[derive(Default)]
+pub struct Parser {
```

Comment:
Nitpick: Could be named deserializer to be symmetric with serializer? Not a blocker, but we might consider renaming this in a _followup_ PR.

Reply:
OK, let's think some more about it. I am wondering what is the idiomatic Rust names.

(We could even name the coder/decoder, from codec)